src/solver/expressions/include/antares/solver/expressions/visitors/AstDOTStyleVisitor.h

```diff
@@ -0,0 +1,196 @@
+    void visit(const Nodes::SubtractionNode* node, std::ostream& os) override;
+    void visit(const Nodes::MultiplicationNode* node, std::ostream& os) override;
+    void visit(const Nodes::DivisionNode* node, std::ostream& os) override;
+    void visit(const Nodes::EqualNode* node, std::ostream& os) override;
+    void visit(const Nodes::LessThanOrEqualNode* node, std::ostream& os) override;
+    void visit(const Nodes::GreaterThanOrEqualNode* node, std::ostream& os) override;
+    void visit(const Nodes::NegationNode* node, std::ostream& os) override;
+    void visit(const Nodes::VariableNode* node, std::ostream& os) override;
+    void visit(const Nodes::ParameterNode* node, std::ostream& os) override;
+    void visit(const Nodes::LiteralNode* node, std::ostream& os) override;
+    void visit(const Nodes::PortFieldNode* node, std::ostream& os) override;
+    void visit(const Nodes::ComponentVariableNode* node, std::ostream& os) override;
+    void visit(const Nodes::ComponentParameterNode* node, std::ostream& os) override;
+
+    /**
+     * @brief Retrieves a unique ID for a given node.
+     *
+     * Generates or retrieves a unique identifier for the specified node.
+     *
+     * @param node The node for which to get the ID.
+     * @return An integer representing the unique ID of the node.
+     */
+    unsigned int getNodeID(const Nodes::Node* node);
+
+    /**
+     * @brief Emits a node to the output stream.
+     *
+     * Writes the DOT representation of a node to the output stream with its associated label
+     * and style.
+     *
+     * @param id The unique ID of the node.
+     * @param label The label to be used for the node.
+     * @param box_style The style to be applied to the node's box.
+     * @param os The output stream to which the node representation is written.
+     */
+    void emitNode(unsigned int id,
+                  const std::string& label,
+                  const BoxStyle& box_style,
+                  std::ostream& os);
+
+    /**
+     * @brief Processes a binary operation node.
+     *
+     * Handles the specific case of binary operation nodes by emitting the appropriate
+     * DOT representation.
+     *
+     * @param node The binary operation node to be processed.
+     * @param label The label to be used for the node.
+     * @param box_style The style to be applied to the node's box.
+     * @param os The output stream to which the node representation is written.
+     */
+    void processBinaryOperation(const Nodes::BinaryNode* node,
+                                const std::string& label,
+                                const BoxStyle& box_style,
+                                std::ostream& os);
+
+    /**
+     * @brief A map of nodes to their unique IDs.
+     *
+     * This map is used to keep track of assigned IDs for each node in the AST.
+     */
+    std::map<std::string, std::map<const Nodes::Node*, unsigned int>> nodeIds_;
```

Comment:
Using pointers as map keys can lead to flaky tests. 

Also, do you need such detailed info ? Why not just map<string, size_t> (type of node -> number of occurrences) ?